src/hera/workflows/workflow_template.py

```diff
@@ -153,5 +162,26 @@ def build(self) -> TWorkflow:
             ),
         )
 
+    def _get_as_workflow(self) -> Workflow:
+        workflow = Workflow(**self.dict())
+        workflow.kind = "Workflow"
+
+        if workflow.name:
+            # In case of a name being > TRUNCATE_LENGTH, we need to truncate the value to assign to generate_name
+            # otherwise the random suffix will make the generated name too long
+            workflow.generate_name = workflow.name[:TRUNCATE_LENGTH] + "-"
+            workflow.name = None
+
+        return workflow
+
+    def create_as_workflow(self, wait: bool = False, poll_interval: int = 5) -> TWorkflow:
```

Comment:
Should generate name be an option here, which gets passed to `get_as_workflow`, perhaps? 🤔 That might give users more control over the behavior of the workflow

Reply:
If you mean as a bool, I did consider this and it would be more flexible, but I don't think there's a use case for _not_ using the generated name as a workflow? Otherwise you'd `create_as_workflow(generate_name=False, ...` and not call it again.

So then if you mean as a string, that sounds more reasonable, optional and we can default it to `workflow.name[:TRUNCATE_LENGTH] + "-"` as before if not given 💯

Reply:
>we can truncate that one

I let the user pass in whatever they want and we don't truncate here which matches normal `Workflow` code - maybe we can add a validator on `generate_name`? Argo will truncate it regardless to add the random suffix if the string is too long

Reply:
> Argo will truncate it regardless
Oh, I see, did not know this

> I let the user pass in whatever they want and we don't truncate here which matches normal Workflow code
This sounds great!

Reply:
>truncate it regardless
Seems I was wrong!

```
argo submit long-name.yaml
ERRO[2023-06-09T15:58:48.404Z] Create request failed                         error="Workflow.argoproj.io \"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaj2vwz\" is invalid: metadata.labels: Invalid value: \"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\": must be no more than 63 characters"
FATA[2023-06-09T15:58:48.404Z] Failed to submit workflow: rpc error: code = InvalidArgument desc = Workflow.argoproj.io "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaj2vwz" is invalid: metadata.labels: Invalid value: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa": must be no more than 63 characters
```

Reply:
Okay no it's weirder than that. If you submit with a generateName > 63 it will error. But if you submit with len(generateName) == 63, Argo will truncate the name to add the suffix... Think it would be helpful to add validators to `Workflow`

Reply:
😅 maybe this is a problem/inconsistency that upstream should fix? Or maybe there's a good reason for it